packages/braid-design-system/src/lib/components/private/Popover/Popover.tsx

```diff
@@ -189,16 +189,25 @@ const PopoverContent = forwardRef<HTMLElement, PopoverProps>(
         return;
       }
 
-      const { top, bottom } = popoverBoundingRect;
-      const distanceFromBottom = window.innerHeight - bottom;
+      const triggerBoundingRect = triggerRef.current?.getBoundingClientRect();
+      if (!triggerBoundingRect) {
+        return;
+      }
+
+      const { height: popoverHeight } = popoverBoundingRect;
 
-      if (placement === 'top' && top < transitionThresholdInPx) {
+      const fitsAbove =
+        triggerBoundingRect.top >= popoverHeight + transitionThresholdInPx;
+      const fitsBelow =
+        window.innerHeight - triggerBoundingRect.bottom >=
+        popoverHeight + transitionThresholdInPx;
```

Comment:
Minor, but a variable that represents the `heightRequired` might make this easier to read.

```suggestion
      const heightRequired = popoverHeight + transitionThresholdInPx;
      const fitsAbove = triggerBoundingRect.top >= heightRequired;
      const fitsBelow =
        window.innerHeight - triggerBoundingRect.bottom >= heightRequired;
```